libsolidity/parsing/Parser.cpp

```diff
@@ -626,10 +630,11 @@ ASTPointer<Mapping> Parser::parseMapping()
 	ASTNodeFactory nodeFactory(*this);
 	expectToken(Token::Mapping);
 	expectToken(Token::LParen);
-	if (!Token::isElementaryTypeName(m_scanner->currentToken()))
-		fatalParserError(string("Expected elementary type name for mapping key type"));
 	ASTPointer<ElementaryTypeName> keyType;
-	keyType = ASTNodeFactory(*this).createNode<ElementaryTypeName>(m_scanner->currentToken());
+	if (ElementaryTypeNameToken::isElementaryTypeName(m_scanner->currentLiteral()))
```

Comment:
Not possible. The token only grabs what it perceives as a token and nothing else from what I can see.